src/dir.targets

```diff
@@ -173,8 +166,31 @@
 
   <!-- On command line run unit tests on CoreCLR after the Test target -->
   <Target Name="RunTestsForProject"
-          AfterTargets="Test"
-          Condition="'$(RunTestsForProject)'=='true'">
+          AfterTargets="CheckTestCategories"
+          Condition="'$(TestDisabled)'!='True'">
+
+    <PropertyGroup>
+      <XunitOptions Condition="'@(RunWithTraits)'!=''">$(XunitOptions) -trait category=@(RunWithTraits, ' -trait category=') </XunitOptions>
+      <XunitOptions Condition="'@(RunWithoutTraits)'!=''">$(XunitOptions) -notrait category=@(RunWithoutTraits, ' -notrait category=') </XunitOptions>
+    </PropertyGroup>
+
+    <!-- xUnit command line without coverage enabled -->
+    <PropertyGroup>
+      <XunitHost>corerun.exe</XunitHost>
+      <TestHost>$(XunitHost)</TestHost>
+      <TestCommandLine>$(XunitCommandLine) $(XunitOptions)</TestCommandLine>
+    </PropertyGroup>
+    
+    <!-- xUnit command line with coverage enabled -->
+    <PropertyGroup Condition="'$(CoverageEnabledForProject)'=='true'">
+      <CoverageHost>$(CoverageToolPath)</CoverageHost>
+      <CoverageOutputFilePath>$(CoverageReportDir)$(MSBuildProjectName).coverage.xml</CoverageOutputFilePath>
+      <CoverageOptions>-filter:"+[*]* -[*.Tests]*" -nodefaultfilters -excludebyattribute:*.ExcludeFromCodeCoverage* -skipautoprops -hideskipped:All -threshold:1</CoverageOptions>
+      <CoverageCommandLine>$(CoverageOptions) -returntargetcode -register:user -target:$(XunitHost) -output:$(CoverageOutputFilePath)</CoverageCommandLine>
+      <TestHost>$(CoverageHost)</TestHost>
+      <XunitOptions>$(XunitOptions) -parallel none</XunitOptions>
+      <TestCommandLine>$(CoverageCommandLine) -targetargs:"$(TestCommandLine)"</TestCommandLine>
+    </PropertyGroup>
```

Comment:
I was able to pull out some of the properties but not all of them. We modify XunitOptions in the target so anything depending on XunitOptions needs to stay